src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java

```diff
@@ -316,6 +258,73 @@ public URLClassLoader run()
         return checkerListener.getResults();
     }
 
+    private void setUpCheckstyleClassloader( Checker checker,
+                                             List<String> classPathStrings,
+                                             List<String> outputDirectories )
+        throws CheckstyleExecutorException
+    {
+        final List<URL> urls = new ArrayList<>( classPathStrings.size() );
+
+        for ( String path : classPathStrings )
+        {
+            try
+            {
+                urls.add( new File( path ).toURI().toURL() );
+            }
+            catch ( MalformedURLException e )
+            {
+                throw new CheckstyleExecutorException( e.getMessage(), e );
+            }
+        }
+
+        for ( String outputDirectoryString : outputDirectories )
+        {
+            try
+            {
+                if ( outputDirectoryString != null )
+                {
+                    File outputDirectoryFile = new File( outputDirectoryString );
+                    if ( outputDirectoryFile.exists() )
+                    {
+                        URL outputDirectoryUrl = outputDirectoryFile.toURI().toURL();
+                        getLogger().debug( "Adding the outputDirectory " + outputDirectoryUrl.toString()
+                                               + " to the Checkstyle class path" );
+                        urls.add( outputDirectoryUrl );
+                    }
+                }
+            }
+            catch ( MalformedURLException e )
+            {
+                throw new CheckstyleExecutorException( e.getMessage(), e );
```

Comment:
Check if this is what you want. It might be, but just to be sure, you want a single bad URL to fail the method even if the other URLs are OK, and before processing the other URLs?